### PR TITLE
Import "types" & "urllib2" on python 3.X  #295

### PR DESCRIPTION
**Pull request by Abhishek-Dev09**
The 'types' module contains a variety of constants to help you determine the type of an object. In Python 3, these constants have been eliminated;just use the primitive type name instead.

The old urllib module in Python 2 had a variety of functions, including urlopen() for fetching data; and splittype(), splithost(), and splituser() forsplitting a URL into its constituent parts. These functions have been reorganized more logically within the new urllib package.
The old urllib2 module in Python 2 has been folded into the urllib package in Python 3. All your urllib2 favorites–the build_opener() method, Request objects, and HTTPBasicAuthHandler and friends–are still available. 

Signed-off-by: Abhishek Kumar <abhishek.kasyap09@gmail.com>

### CODE REVIEW

**Comment by Abhishek-Dev09** [comment]
If anyone find more suitable title and msg for this commit.Suggest it

File: src/commoncode/functional.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 from __future__ import absolute_import, print_function
 
 import functools
-from types import ListType, TupleType, GeneratorType
+from types import *
```
Comment (pombredanne) [inline src/commoncode/functional.py]: Please never ever use `import *` but always list what you import explicitly

File: src/scancode/__init__.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 from os.path import getmtime
 from os.path import join
 from os.path import exists
-from types import BooleanType
+from types import *
```
Comment (pombredanne) [inline src/scancode/__init__.py]: No * please

File: tests/licensedcode/data/slic-tests/identification/repo [hunk 3]
```diff
@@ -123,7 +123,10 @@ if sys.version_info[0] == 3:
   import urllib.error
 else:
   import imp
-  import urllib2
+  try:
```
Comment (pombredanne) [inline tests/licensedcode/data/slic-tests/identification/repo]: This is a test data file that is NOT part of the code we run, just something we test license detection on... so this file should not be modified at all.

File: src/commoncode/functional.py [hunk 4]
```diff
@@ -25,7 +25,15 @@
...
-from types import ListType, TupleType, GeneratorType
+
+# Python 2 and 3 support
+try:
+    # Python 2
+    from types import ListType, TupleType, GeneratorType
+except ImportError, NameError:
+    from types import GeneratorType
+    ListType = list
+    TupleType = tuple
```
Comment (pombredanne) [inline src/commoncode/functional.py]: that's clean... but could we instead just use list.tuple everywhere and only import GeneratorType on both py2 and 3?

File: src/commoncode/functional.py [hunk 5]
```diff
@@ -53,7 +61,7 @@ def flatten(seq):
     """
     r = []
     for x in seq:
-        if isinstance(x, (ListType, TupleType)):
+        if isinstance(x, (ListType, TupleType, array)):
```
Comment (pombredanne) [inline src/commoncode/functional.py]: Why did you add array to this? Aldo with my other comment above, using `(list, tuple,)` would be enough IMHO

File: src/scancode/__init__.py [hunk 6]
```diff
@@ -35,7 +35,14 @@
...
 from os.path import exists
-from types import BooleanType
+
+# Python 2 and 3 support
+try:
+    #Python2
+    from types import BooleanType
+except NameError:
+    # Python 3
+    BooleanType = bool
```
Comment (pombredanne) [inline src/scancode/__init__.py]: I think it is simpler to use `bool` on both versions.: no import needed anymore

File: src/commoncode/functional.py [hunk 7]
```diff
@@ -25,7 +25,15 @@
 from __future__ import absolute_import, print_function
 
 import functools
-from types import ListType, TupleType, GeneratorType
+
+# Python 2 and 3 support
```
Comment (singh1114) [inline src/commoncode/functional.py]: No need for this comment.

File: src/commoncode/functional.py (hunk 5 above)
Comment (singh1114) [inline src/commoncode/functional.py]: Why have you included array as well?

File: src/scancode/__init__.py [hunk 8]
```diff
@@ -35,7 +35,14 @@
 from os.path import getmtime
 from os.path import join
 from os.path import exists
-from types import BooleanType
+
+# Python 2 and 3 support
```
Comment (singh1114) [inline src/scancode/__init__.py]: Again no need for such comments.
